client/my-sites/checkout/index.js

```diff
@@ -29,6 +29,8 @@ export default function () {
 	if ( isLoggedOut ) {
 		page( '/checkout/offer-quickstart-session', upsellNudge, makeLayout, clientRender );
 
+		page( '/checkout/no-site/:lang?', noSite, checkout, makeLayout, clientRender );
```

Comment:
You can use [getLanguageRouteParam](https://github.com/Automattic/wp-calypso/blob/master/client/lib/i18n-utils/utils.js#L107) here instead. It will an extra layer of specificity.